test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Smart/SmartSearchTests.cs

```diff
@@ -748,5 +752,278 @@ private void ConfigureFhirRequestContext(
 
             contextAccessor.RequestContext.AccessControlContext.Returns(accessControlContext);
         }
+
+        // SMART v2 Granular Scope Tests
+
+        [SkippableFact]
+        public async Task GivenSmartV2CreateScope_WhenCreatingPatient_ThenPatientIsCreated()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            var scopeRestriction = new ScopeRestriction("Patient", Core.Features.Security.DataActions.Create, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction });
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentId = "smart-patient-test";
+            _contextAccessor.RequestContext.AccessControlContext.CompartmentResourceType = "Patient";
+
+            var newPatient = new Patient
+            {
+                Id = "smart-v2-create-test",
+                Name = new List<HumanName> { new HumanName().WithGiven("TestCreate").AndFamily("SmartV2") },
+            };
+
+            var result = await PutResource(newPatient);
+            Assert.NotNull(result);
```

Comment:
should we check for success status code?

Reply:
I checked for the resource id in the result.